README.md

```diff
@@ -1,2 +1,78 @@
-# jboss-eap-openshift-templates
-ImageStreams and Templates for the OpenShift Images of JBoss EAP
+# Red Hat JBoss Enterprise Application Platform XP 1.0 OpenShift container image imagestreams and templates
+
+## Importing Imagestreams and Templates
+
+You should have the 'oc' tools available and be logged in to your OpenShift instance. For more details on how to do this, please consult the OpenShift documentation.
+For example, for OpenShift Online, see: https://docs.openshift.com/online/cli_reference/get_started_cli.html
+[source, bash]
+----
+oc replace --force -f https://raw.githubusercontent.com/jboss-container-images/jboss-eap-openshift-templates/eap-xp1/jboss-eap-xp1-openjdk11-openshift.json
+
+for resource in eap-xp1-amq-s2i.json \
+  eap-xp1-basic-s2i.json \
+  eap-xp1-third-party-db-s2i.json
+do
+  oc replace --force -f https://raw.githubusercontent.com/jboss-container-images/jboss-eap-openshift-templates/eap-xp1/templates/${resource}
+done
+----
+
+If you have administrative access to the general `openshift` namespace and want the image streams and templates to be accessible by all projects, add -n openshift to the oc replace line of the command. For example:
+
+[source, bash]
+----
+oc replace -n openshift --force -f ...
+----
+
+#### Updating Existing Images
+To update an to the most recent image:
+
+[source, bash]
+----
+oc import-image jboss-eap-xp1-openjdk11-openshift:1.0
+oc import-image jboss-eap-xp1-openjdk11-runtime-openshift:1.0
+----
+
+Optionally include namespace to the import:
+[source, bash]
+----
+oc -n myproject import-image jboss-eap-xp1-openjdk11-openshift:1.0
+oc -n myproject import-image jboss-eap-xp1-openjdk11-runtime-openshift:1.0
+----
+
+#### Creating New Applications With Templates
+Example:
+
+[source, bash]
+----
+oc -n myproject new-app eap-xp1-basic-s2i -p APPLICATION_NAME=eap-xp1-basic-app
+----
+
+or, if a namespace was used with `-n` above:
+[source, bash]
+----
+oc -n myproject new-app eap-xp1-basic-s2i -p APPLICATION_NAME=eap-xp1-basic-app -p IMAGE_STREAM_NAMESPACE=myproject
+----
+
+## Creating Secrets
+
+Some of the included templates require the creation of secrets.
+
+Example:
+[source, bash]
+----
+$ keytool -genkey -keyalg RSA -alias eapdemo-selfsigned -keystore keystore.jks -validity 360 -keysize 2048
+$ oc secrets new eap7-app-secret keystore.jks
+----
+
+Example secrets may also be found here: https://github.com/jboss-openshift/application-templates/tree/master/secrets
+
+[source, bash]
+----
+oc create -n myproject -f https://raw.githubusercontent.com/jboss-openshift/application-templates/master/secrets/eap-app-secret.json
```

Comment:
should we move some of these secrets to this repository?

Reply:
We can, but rather than duplicate them over every branch, perhaps we could just have them in master, then reference them from the other branches?

If at some point we want to regenerate these secrets due to expiration or something, it will be easier if they're all in one place.